src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -191,64 +185,86 @@
     hosts catalog changes in accordance to the users preference)
 
   * Submit the resulting catalog"
-  [{:keys [host lastrun catalog report factset base-url run-interval rand-percentage] :as state} clock]
+  [{:keys [host lastrun catalog report factset] :as state} base-url run-interval rand-percentage clock]
   (if-not (> (- clock lastrun) run-interval)
     state
-    (let [catalog (some-> catalog update-catalog (maybe-tweak-catalog rand-percentage))
-          report (some-> report update-report-run-fields)
-          factset (some-> factset (update-factset rand-percentage))]
+    (let [catalog (some->> catalog (maybe-tweak-catalog rand-percentage))
+          factset (some->> factset (maybe-tweak-factset rand-percentage))]
       ;; Submit the catalog and reports in separate threads, so as to not
       ;; disturb the world-loop and otherwise distort the space-time continuum.
       (when catalog
         (future
           (try
-            (client/submit-catalog base-url 6 (json/generate-string catalog))
+            (->> catalog
+                 update-producer-timestamp
+                 json/generate-string
+                 (client/submit-catalog base-url 6))
             (log/infof "[%s] submitted catalog" host)
             (catch Exception e
               (log/errorf "[%s] failed to submit catalog: %s" host e)))))
       (when report
         (future
           (try
-            (client/submit-report base-url 5 (json/generate-string report))
+            (some->> report
+                     update-report
+                     json/generate-string
+                     (client/submit-report base-url 5))
             (log/infof "[%s] submitted report" host)
             (catch Exception e
               (log/errorf "[%s] failed to submit report: %s" host e)))))
       (when factset
         (future
           (try
-            (client/submit-facts base-url 4 (json/generate-string factset))
+            (->> factset
+                 update-producer-timestamp
+                 json/generate-string
+                 (client/submit-facts base-url 4)) 
             (log/infof "[%s] submitted factset" host)
             (catch Exception e
               (log/errorf "[%s] failed to submit factset: %s" host e)))))
       (assoc state
-        :lastrun clock
-        :catalog catalog))))
+             :lastrun clock
+             :factset factset
+             :catalog catalog))))
```

Comment:
I see why now -- the report update doesn't do any mutation or take rp into effect at all. That's a bug, pretty sure there's an existing ticket.